### PR TITLE
Don/up/filter

### PR DESCRIPTION
**Pull request by DonIsaac**
- Async stuff
- Lots o'testing

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -33,9 +33,36 @@ jobs:
...
         with:
           name: cmsc435-discord-bot
           path: dist/**
           retention-days: 1
+
+  test:
+    name: Run Tests
+    runs-on: ubuntu-latest
+    env:
+      WEBHOOK_URL: https://discord.com/api/webhooks/800453941313745987/foo-bar-baz-bang
```
Comment (DanRowe) [inline .github/workflows/workflow.yml:47]: We can use ${{ secrets.WEBHOOK_URL}} to keep this hidden

File: src/index.ts [hunk 2]
```diff
@@ -12,54 +13,63 @@ const { blogUrl, webhookUrl, blogFile } = config
 // const blogUrl = 'https://seam.cs.umd.edu/purtilo/435/blog.html'
 // const webhookUrl = process.env.WEBHOOK_URL || ''
 
-const blogs = fs.readFileSync(blogFile, 'utf8').split('\n')
+// const blogs = fs.readFileSync(blogFile, 'utf8').split('\n')
+
+const loadBlogFile = async (): Promise<string[]> => {
+    try {
+        const s = await fs.stat(blogFile)
+        // if (s.isDirectory()) throw new Error('Blog file is a directory')
```
Comment (DanRowe) [inline src/index.ts]: Is this commented out line needed? If not, remove useless assignment to `s`.

File: src/index.ts [hunk 3]
```diff
@@ -12,54 +13,63 @@ const { blogUrl, webhookUrl, blogFile } = config
 // const blogUrl = 'https://seam.cs.umd.edu/purtilo/435/blog.html'
```
Comment (DanRowe) [inline src/index.ts]: remove commented out code

File: src/blogsToday [hunk 4]
```diff
@@ -0,0 +1,2 @@
+
+2020/01/29a
```
Comment (DanRowe) [inline src/blogsToday]: keep this file empty

**Review by DanRowe — CHANGES_REQUESTED** [review]
Wow very thorough tests. Just some minor cleanup before we get this merged in.

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (DonIsaac) [reply .github/workflows/workflow.yml:47]: We can, but that's a mock URL. I don't think it's necessary

**Pull request merged**